src/pages/audio-rewards-page/components/ClaimRewardButton.tsx

```diff
@@ -168,67 +169,73 @@ const ClaimRewardButton = ({
-          break
-        default:
-          // nothing
-          break
-      }
-    })
+      flow.on('ui', (event: FlowUIOpenEvent | FlowUICloseEvent) => {
+        switch (event.action) {
+          case 'opened':
+            dispatch(setCognitoFlowStatus({ status: CognitoFlowStatus.OPENED }))
+            break
+          case 'closed':
+            console.error(
+              'Error claiming reward: user closed the cognito flow modal'
+            )
+            dispatch(setCognitoFlowStatus({ status: CognitoFlowStatus.CLOSED }))
+            dispatch(setClaimStatus({ status: ClaimStatus.ERROR }))
+            break
+          default:
+            // nothing
+            break
+        }
+      })
 
-    flow.on('session', (event: FlowSessionEvent) => {
-      switch (event.action) {
-        case 'passed':
-          console.info(
-            'User successfully completed their flow session, trying reward claim again...'
-          )
-          flow.close()
-          retryClaimReward()
-          break
-        case 'created':
-          console.info('User started a new flow session')
-          break
-        case 'resumed':
-          console.info('User resumed an existing flow session')
-          break
-        case 'failed':
-          console.error('Error claiming reward: User failed their flow session')
-          flow.close()
-          dispatch(setClaimStatus({ status: ClaimStatus.ERROR }))
-          break
-        default:
-          // nothing
-          break
-      }
-    })
+      flow.on('session', (event: FlowSessionEvent) => {
+        switch (event.action) {
+          case 'passed':
+            console.info(
+              'User successfully completed their flow session, trying reward claim again...'
+            )
+            flow.close()
+            retryClaimReward()
+            break
+          case 'created':
+            console.info('User started a new flow session')
+            break
+          case 'resumed':
+            console.info('User resumed an existing flow session')
+            break
+          case 'failed':
+            console.error(
+              'Error claiming reward: User failed their flow session'
+            )
+            flow.close()
+            dispatch(setClaimStatus({ status: ClaimStatus.ERROR }))
+            break
+          default:
+            // nothing
+            break
+        }
+      })
 
-    flow.on('error', (event: FlowErrorEvent) => {
-      console.error(`Error claiming reward: Flow error! ${event.message}`)
-      dispatch(setClaimStatus({ status: ClaimStatus.ERROR }))
-    })
+      flow.on('error', (event: FlowErrorEvent) => {
+        console.error(`Error claiming reward: Flow error! ${event.message}`)
+        dispatch(setClaimStatus({ status: ClaimStatus.ERROR }))
+      })
 
-    flow.open()
+      flow.open()
+    } catch (e) {
+      console.error('COGNITO: Failed to get signature.', e)
```

Comment:
just wondering if there is a way to minimize the try/catch block here since its really just for the getCognitoSignature. potentially that function just throws the updated error with a console.log?